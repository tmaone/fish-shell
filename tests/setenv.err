too many arguments test
setenv: Too many arguments
