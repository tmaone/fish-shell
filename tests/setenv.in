# Verify the correct behavior of the `setenv` compatibility shim.

# No args to `setenv` should emit the current set of env vars. The first two
# commands verify that `setenv` does not report non-env vars.
set -g setenv1 abc
setenv | grep '^setenv1=$'
set -gx setenv1 xyz
setenv | grep '^setenv1=xyz$'

# A single arg should set and export the named var to nothing.
setenv setenv2
env | grep '^setenv2=$'

# Three or more args should be an error.
echo too many arguments test >&2
setenv var hello you

# Two args should set the named var to the second arg
setenv setenv3 'hello you'
setenv | grep '^setenv3=hello you'
