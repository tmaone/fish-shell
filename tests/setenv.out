setenv1=xyz
setenv2=
setenv3=hello you
